js/setpassword.js

```diff
@@ -0,0 +1,61 @@
+(function () {
+	var SetPassword = {
+		init : function() {
+			$('#set-password #submit').click(this.onClickSetPassword);
+		},
+
+		onClickSetPassword : function(event){
+			event.preventDefault();
+			var passwordObj = $('#password');
+			if (passwordObj.val()){
+				$.post(
+					passwordObj.parents('form').attr('action'),
+					{password : passwordObj.val()}
+				).done(function (result) {
+					OCA.UserManagement.SetPassword._resetDone(result);
+				}).fail(function (result) {
+					OCA.UserManagement.SetPassword._onSetPasswordFail(result);
+				});
+			}
+		},
+
+		_onSetPasswordFail: function(result) {
+			var responseObj = JSON.parse(result.responseText);
+			var errorObject = $('#error-message');
+			var showErrorMessage = false;
+
+			var errorMessage;
+			errorMessage = responseObj.message;
+
+			if (errorMessage) {
+				errorObject.text(errorMessage);
+				errorObject.show();
+				$('#submit').prop('disabled', true);
+			}
+		},
+
+		_resetDone : function(result){
+			if (result && result.status === 'success') {
+				$.post(
+					OC.webroot + '/',
+					{
+						user : window.location.href.split('/').pop(),
```

Comment:
this is not unit testable, use OC.Host(), etc which are stubbable/mockable

Reply:
Removed the ajax request. The OC.redirect is sufficient, imho.